app/models/manageiq/providers/ibm_cloud/vpc/cloud_manager/provision_workflow/volumes.rb

```diff
@@ -0,0 +1,140 @@
+# frozen_string_literal: true
+
+# Contains the elements used to populate and validate boot volume, attached volumes and new volume related fields.
+module ManageIQ::Providers::IbmCloud::VPC::CloudManager::ProvisionWorkflow::Volumes
+  # These are the keys that are populated in the Volumes tab of the UI.
+  # @return [Array<String>]
+  def volume_dialog_keys
+    %i[volume_name volume_profile volume_size volume_on_instance_delete].freeze
+  end
+
+  # Fetch volume profiles from inventory.
+  # @param _options [void]
+  # @return [Hash] Hash with ems_ref as key and name as value.
+  def storage_type_to_profile(_options = {})
+    @storage_type_to_profile ||= string_dropdown(ar_ems.cloud_volume_types, :remove_fields => %w[custom])
+  rescue => e
+    logger(__method__).log_backtrace(e)
+  end
+
+  # Wait until zone is set, then fetch volumes that have a status of available and reside in that zone.
+  # @param _options [void]
+  # @return [Hash] Hash with ems_ref as key and name as value.
+  def cloud_volumes_to_volumes(_options = {})
+    zone = field(:placement_availability_zone)
+    return {} if zone.nil?
+
+    ar_volumes = ar_ems.cloud_volumes.select do |cloud_volume|
+      cloud_volume[:status] == 'available' && cloud_volume[:availability_zone_id] == zone
+    end
+    string_dropdown(ar_volumes)
+  rescue => e
+    logger(__method__).log_backtrace(e)
+  end
+
+  # Perform any needed manipulation and validation for new volume fields.
+  # @param values [Hash<Symbol, String>] Hash of all the keys and values submitted in the form.
+  # @return [void]
+  def parse_new_volumes_fields(values)
+    values[:new_volumes] = create_new_volumes_array(values)
+    validate_volumes(values[:new_volumes])
+  end
+
+  # Select keys that have one of the volume dialog keys followed by a digit.
+  # @param values [Hash<Symbol, String>] Hash of all the keys and values submitted in the form.
+  # @return [Array<Hash{Symbol => String, Integer}>]
+  def create_new_volumes_array(values)
+    new_volumes = []
+    # Regex to place the field name in the field index and count in the count index.
+    keys_regex = /(?<field>#{volume_dialog_keys.join('|')})_(?<count>\d+)/
+
+    values.each do |key, value|
+      next if value.nil? || (value.respond_to?(:length) && value.length.zero?)
+
+      reg = key.match(keys_regex)
+      next if reg.nil?
+
+      count = reg[:count].to_i
+      new_volumes[count] ||= {}
+
+      field = reg[:field].to_sym
+
+      # Use integer if value starts and ends with digits.
+      new_volumes[count][field] = value.match?('^\d+$') ? value.to_i : value
```

Comment:
Super-minor, but prefer a regex here over a string with a regex in it.  Performance-wise, the former is significantly faster (though admittedly that's unlikely an actual problem here, but it's more for code consistency)

Reply:
Same elsewhere, such as on line 107

Reply:
I read the same thing when I was figuring this out. I prefer match because it is googable for people not familiar with the `=~` which aids in maintenance. I'm also not comfortable with it and don't want to do a bunch of testing for something so minor.